File: BI-TJV/semestral/app-model/src/main/java/wrzecond/order_food/TjvOrderFoodDTO.java
package wrzecond.order_food;

import lombok.AccessLevel;
import lombok.AllArgsConstructor;
import lombok.EqualsAndHashCode;
import lombok.Getter;

import java.sql.Timestamp;

@EqualsAndHashCode
@AllArgsConstructor(access = AccessLevel.PROTECTED)
abstract class TjvOrderFoodDTO {

    // === PROPERTIES ===
    protected final Integer id;
    @Getter private final Integer order;

    @Getter private final Timestamp datetime;
    @Getter private final Integer count;

}

File: BI-KOT/semestral/app-model/src/main/kotlin/wrzecond/TjvOrderFoodDTO.kt
package wrzecond

import java.sql.Timestamp

data class TjvOrderFoodReadDTO (
    override val id: Int,
    val order: Int,
    val food: TjvFoodReadDTO,
    val datetime: Timestamp,
    val count: Int
) : IReadDTO { override val displayName = "$order - ${food.displayName}" }

data class TjvOrderFoodCreateDTO (
    val order: Int,
    val food: Int,
    val datetime: Timestamp,
    val count: Int
) : ICreateDTO

data class TjvOrderFoodUpdateDTO (
    val order: Int? = null,
    val food: Int? = null,
    val datetime: Timestamp? = null,
    val count: Int? = null
) : IUpdateDTO